Use read-string if .clj file when parsing cli opts

frankitox opened the pull request:
I'm trying to fix the problem described in #190. It's just a couple of lines of code. Let me know what you think.

weavejester commented on cljfmt/src/cljfmt/diff.clj:
```diff
@@ -16,6 +16,9 @@
        ^java.io.File (apply io/file)
        .getCanonicalPath))
 
+(defn filename-ext [filename]
```
Could you move this function to main.clj (and make it private)? It doesn't fit in diff.clj.

weavejester commented on cljfmt/src/cljfmt/main.clj:
```diff
@@ -136,8 +136,12 @@
 
-(def ^:private cli-file-reader
-  (comp edn/read-string slurp diff/to-absolute-path))
+(defn- cli-file-reader [opt-value]
```
Instead of `opt-value`, prefer a more descriptive name like `filepath`.

weavejester commented on cljfmt/src/cljfmt/main.clj:
```diff
@@ -136,8 +136,12 @@
-(def ^:private cli-file-reader
-  (comp edn/read-string slurp diff/to-absolute-path))
+(defn- cli-file-reader [opt-value]
+  (let [filename (diff/to-absolute-path opt-value)
```
What's the purpose of converting the filename to an absolute path?

frankitox replied:
Well, I just blindly copied `cli-file-reader`'s source code. I made a slight [change](https://github.com/weavejester/cljfmt/pull/202/commits/ffc2dcc11eec599a5b6abb255b5e57330722c17f#diff-a57e762da42583224b62656b3ae05addR144-R145), so `filename-ext` receives the raw string, but let me know if you meant something else.

weavejester replied:
Ah, I see. I'm not sure why the `diff/to-absolute-path` is there to begin with. I think we can safely remove it.

frankitox replied:
Great! Yes, I can't see a reason for it either. I just [removed it](https://github.com/weavejester/cljfmt/pull/202/commits/334ef64029fb132249b89f7f06af3d8aa0d76a8f).

frankitox commented:
Hi @weavejester, thanks for taking the time to review this! These are the changes I just made:

- [x] Moved `relative-path` to `cljfmt.main` and make it private (let me know if you prefer other name for `relative-path`).
- [x] Renamed `opt-value` to `filepath`.

Note that `cli-file-reader` is also used by `--alias-map`. Which means you can also pass `.edn` files to that parameter. Is that ok with you?

weavejester commented:
It seems fine to be consistent and have all options support both .edn and .clj file types.

frankitox commented:
Gotcha. Give me a moment because I force pushed instead of creating a new commit, so you can actually see what I changed.

frankitox commented:
There we go, you can check the [last changes](https://github.com/weavejester/cljfmt/pull/202/commits/ffc2dcc11eec599a5b6abb255b5e57330722c17f) now. Thanks!

weavejester commented:
Thanks! I also had a question on why you're using `diff/to-absolute-path` in `cli-file-reader`. It doesn't look necessary to me.
